client/signup/steps/woocommerce-install/transfer.tsx

```diff
@@ -10,6 +22,76 @@ interface Props {
 export default function Transfer( { goToStep }: Props ): ReactElement | null {
 	const { __ } = useI18n();
 
+	const siteId = useSelector( getSelectedSiteId ) as number;
+
+	const [ progress, setProgress ] = useState( 0 );
+	const [ error, setError ] = useState( { transferFailed: false, transferStatus: null } );
+
+	const dispatch = useDispatch();
+
+	// Select state
+	const fetchingTransferStatus = !! useSelector( ( state: AppState ) =>
+		isFetchingAutomatedTransferStatus( state, siteId )
+	);
+	const transferStatus = useSelector( ( state: AppState ) =>
+		getAutomatedTransferStatus( state, siteId )
+	);
+
+	const transferFailed = useSelector( ( state: AppState ) => hasUploadFailed( state, siteId ) );
+
+	// Initiate Atomic transfer
+	useEffect( () => {
+		if ( ! siteId ) {
+			return;
+		}
+
+		dispatch( fetchAutomatedTransferStatus( siteId ) ); // todo: can possibly drop this, polling should be kicked off by initiate call
+		dispatch( initiateThemeTransfer( siteId, null, 'woocommerce' ) );
+	}, [ siteId, dispatch ] );
+
+	// Watch transfer status
+	useEffect( () => {
+		if ( fetchingTransferStatus ) {
+			return;
+		}
+
+		switch ( transferStatus ) {
+			case transferStates.NONE:
+			case transferStates.PENDING:
+			case transferStates.INQUIRING:
+			case transferStates.PROVISIONED:
+			case transferStates.FAILURE:
+			case transferStates.START:
+			case transferStates.REVERTED:
+				setProgress( 20 );
+				break;
+			case transferStates.SETUP:
+			case transferStates.CONFLICTS:
+			case transferStates.ACTIVE:
+				setProgress( 50 );
+				break;
+			case transferStates.UPLOADING:
+			case transferStates.BACKFILLING:
+				setProgress( 60 );
+				break;
+			case transferStates.COMPLETE:
+				setProgress( 100 );
+				goToStep( 'complete' );
+				break;
+		}
+
+		if (
+			transferFailed ||
+			transferStatus === transferStates.ERROR ||
+			transferStatus === transferStates.FAILURE ||
+			transferStatus === transferStates.REQUEST_FAILURE ||
+			transferStatus === transferStates.CONFLICTS
+		) {
+			setProgress( 100 );
+			setError( { transferFailed, transferStatus } );
+		}
+	}, [ setProgress, goToStep, fetchingTransferStatus, transferStatus, transferFailed ] );
+
```

Comment:
Should we check whether the site ID is not defined? Pretty sure it is a really unlikely situation, but maybe it is worth it:

```es6
if ( ! siteId ) {
  return null;
}
```

Reply:
Probably a good idea to be defensive. siteSlug is ensured by singup framework dependencies which sets siteId but I'm not too sure how solid that code is.